index.js

```diff
@@ -20,19 +28,55 @@ AFRAME.registerComponent('simple-link', {
   /**
    * Called once when component is attached. Generally for initial setup.
    */
-  init() {},
+  init() {
+    this.navigate = this.navigate.bind(this);
+    const { el } = this;
+
+    el.setAttribute('geometry', {
+      primitive: 'circle',
+      radius: 1
+    });
+    if (this.data.image) {
+      el.setAttribute('material', {
+        src:
+          typeof this.data.image === 'string'
+            ? this.data.image
+            : this.data.image.src,
+        color: this.data.color
+      });
+    }
+
+    const textEl = this.textEl || document.createElement('a-entity');
```

Comment:
Interesting, under what circumstances would `this.textEl` have a value? `init()` should only be called once, when the component is initializing, so I'm thinking `this.textEl` might never exist at this point. I also don't see any code _setting_ `this.textEl`, or using `this.textEl`, so I think it might make sense to just create an entity here.

Reply:
Agreed. This was inherited from whatever Mike started, and yes it does look weird with no possible source for `textEl`... except perhaps one nested as a child? Still doesn't make sense. Removed.